### PR TITLE
Added shell scripts to create and configure OKE cluster and run integration test

### PR DESCRIPTION
**Pull request by marinakog**
https://build.weblogick8s.org:8443/job/wko-oke-test/166/

### CODE REVIEW

File: kubernetes/samples/scripts/terraform/oke.create.sh [hunk 1]
```diff
@@ -37,12 +37,13 @@ function setupTerraform () {
       curl -O https://releases.hashicorp.com/terraform/0.11.10/terraform_0.11.10_darwin_amd64.zip
       unzip terraform_0.11.10_darwin_amd64.zip
     elif [[ "${OSTYPE}" == "linux"* ]]; then
-       curl -O https://releases.hashicorp.com/terraform/0.11.8/terraform_0.11.8_linux_amd64.zip
-       unzip terraform_0.11.8_linux_amd64.zip
+#setup terraform#curl -LO --retry 3 https://releases.hashicorp.com/terraform/0.11.8/terraform_0.11.8_linux_amd64.zip#unzip -o terraform_0.11.8_linux_amd64.zip -d ${WORKSPACE}/bin/#rm terraform_0.11.8_linux_amd64.zip#chmod +x ${WORKSPACE}/bin/terraform
```
Comment (xiancao) [inline kubernetes/samples/scripts/terraform/oke.create.sh:40]: Can you put this to multiple lines or just leave "#setup terraform"?

File: kubernetes/samples/scripts/terraform/oke.create.sh [hunk 2]
```diff
@@ -68,6 +69,60 @@ function createCluster () {
     terraform apply -auto-approve -var-file=${terraformVarDir}/${clusterTFVarsFile}.tfvars
 }
 
+function createRoleBindings () {
+kubectl -n kube-system create serviceaccount $okeclustername-sa
```
Comment (xiancao) [inline kubernetes/samples/scripts/terraform/oke.create.sh:73]: add some indentation for the function body.

File: kubernetes/samples/scripts/terraform/oke.create.sh [hunk 3]
```diff
@@ -68,6 +69,60 @@ function createCluster () {
...
+kubectl create clusterrolebinding add-on-cluster-admin --clusterrole=cluster-admin --serviceaccount=kube-system:$okeclustername-sa
+TOKENNAME=`kubectl -n kube-system get serviceaccount/$okeclustername-sa -o jsonpath='{.secrets[0].name}'`
+TOKEN=`kubectl -n kube-system get secret $TOKENNAME -o jsonpath='{.data.token}'| base64 --decode`
+kubectl config set-credentials $okeclustername-sa --token=$TOKEN
+kubectl config set-context --current --user=$okeclustername-sa
+}
+
+function checkClusterRunning () {
+
+echo 'Confirm we have kubectl working...'
```
Comment (xiancao) [inline kubernetes/samples/scripts/terraform/oke.create.sh:83]: add some indentation for the function body.

File: kubernetes/samples/scripts/terraform/oke.delete.sh [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+set -o errexit
+set -o pipefail
+
+function prop {
+    grep "${1}" ${oci_property_file}| grep -v "#" | cut -d'=' -f2
+}
+
+function cleanupLB {
+  echo 'Clean up left over LB'
+myvcn_id=`oci network vcn list --compartment-id $compartment_ocid  --display-name=${clusterName}_vcn | jq -r '.data[] | .id'`
```
Comment (xiancao) [inline kubernetes/samples/scripts/terraform/oke.delete.sh:18]: add some indentation for the function body.

File: oketest.sh [hunk 5]
```diff
@@ -0,0 +1,194 @@
+#!/bin/bash
+# Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+#
+# This script provisions a OKE Kubernetes cluster using terraform (https://www.terraform.io/) and runs the new
+# integration test suite against that cluster. Blog https://blogs.oracle.com/weblogicserver/easily-create-an-oci-container-engine-for-kubernetes-cluster-with-terraform-installer-to-run-weblogic-server
+# provides detailed explanation for OCI properties file creation.
+#
+#
```
Comment (xiancao) [inline oketest.sh:9]: delete this line

File: oketest.sh [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+    ;;
+    *) usage 1
+    ;;
+  esac
+done
+
+k8s_version=$(prop 'k8s.version')
+echo "Using Kubernetes version: ${k8s_version}"
+
+
```
Comment (xiancao) [inline oketest.sh:99]: delete this empty line

File: oketest.sh [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+echo 'Create a OKE cluster'
+mkdir -p "${WORKSPACE}/terraform"
+cp -rf ${terraform_script_dir_name}/*.tf ${WORKSPACE}/terraform/.
+cp -rf ${WORKSPACE}/kubernetes/samples/scripts/terraform/template.tfvars ${WORKSPACE}/terraform/.
+cp -rf ${WORKSPACE}/kubernetes/samples/scripts/terraform/*.sh ${WORKSPACE}/terraform/.
+chmod 777 ${WORKSPACE}/terraform/*.sh
+mkdir -p ${WORKSPACE}/terraform/terraforminstall
+
+
```
Comment (xiancao) [inline oketest.sh:127]: delete this empty line

File: oketest.sh [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+if ! sh ${WORKSPACE}/terraform/oke.create.sh ${oci_property_file} ${WORKSPACE}/terraform ; then
+sh ${WORKSPACE}/terraform/oke.delete.sh ${oci_property_file} ${WORKSPACE}/terraform
```
Comment (xiancao) [inline oketest.sh:129]: add some indentation here.

File: oketest.sh [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+provisioner: oracle.com/oci-fss
+parameters:
+  # Insert mount target from the FSS here
+  mntTargetId: ${mount_target_id}
+EOF
+
+echo 'Set up test running ENVVARs...'
+NODE_IP=`kubectl get nodes -o wide| awk '{print $7}'| tail -n+3`
+if [ -z "$NODE_IP" ]; then
+	echo "retry get node ip ";
```
Comment (xiancao) [inline oketest.sh:165]: use same indentation in this section.

File: oketest.sh [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+
+echo 'Clean up result root...'
+rm -rf "${RESULT_ROOT:?}/*"
+
+cd ${WORKSPACE}
+
+echo 'Run tests...'
+
+if [ "${maven_profile_name}" = "oke-cert" ]; then
+echo "Running mvn -Dwdt.download.url=${wdt_download_url} -Dwit.download.url=${wit_download_url} -pl integration-tests -P ${maven_profile_name} verify"
```
Comment (xiancao) [inline oketest.sh:181]: add indentation in this section

File: oketest.sh [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+mvn -Dwdt.download.url="${wdt_download_url}" -Dwit.download.url="${wit_download_url}" -Djdk.tls.client.protocols=TLSv1.2 -pl integration-tests -P ${maven_profile_name} verify 2>&1 | tee "${RESULT_ROOT}/oke.log"
+else
+  if [ "${parallel_run}" = "false" ]; then
+  echo "Running mvn -Dit.test=${test_filter} -Dwdt.download.url=${wdt_download_url} -Dwit.download.url=${wit_download_url} -pl integration-tests -P integration-tests verify"
```
Comment (xiancao) [inline oketest.sh:185]: add indentation in this section

File: oketest.sh [hunk 12]
```diff
@@ -0,0 +1,194 @@
...
+  mvn -Dit.test="${test_filter}, !ItExternalRmiTunneling, !ItSamples, !ItMiiSample, !ItTwoDomainsLoadBalancers, !ItMonitoringExporter, !ItPodRestart" -Dwdt.download.url="${wdt_download_url}" -Dwit.download.url="${wit_download_url}" -Djdk.tls.client.protocols=TLSv1.2 -pl integration-tests -P ${maven_profile_name} verify 2>&1 | tee "${RESULT_ROOT}/oke.log"
+  else
+    echo "Running mvn -Dit.test=${test_filter}, !ItOperatorUpgrade, !ItDedicatedMode -Dwdt.download.url=${wdt_download_url} -Dwit.download.url=${wit_download_url} -DPARALLEL_CLASSES=${parallel_run} -DNUMBER_OF_THREADS=${threads}  -pl integration-tests -P ${MVN_PROFILE} verify"
+     mvn -Dit.test="${test_filter}, !ItTwoDomainsLoadBalancers, !ItExternalRmiTunneling, !ItMiiSample, !ItSamples, !ItPodRestart" -Dwdt.download.url="${wdt_download_url}" -Dwit.download.url="${wit_download_url}" -DPARALLEL_CLASSES="${parallel_run}" -DNUMBER_OF_THREADS="2" -Djdk.tls.client.protocols=TLSv1.2 -pl integration-tests -P ${maven_profile_name} verify 2>&1 | tee "${RESULT_ROOT}/oke.log"
+  fi
+fi
+
```
Comment (xiancao) [inline oketest.sh:192]: delete these empty lines.

File: kubernetes/samples/scripts/terraform/oke.create.sh (hunk 1 above)
Comment (marinakog) [reply kubernetes/samples/scripts/terraform/oke.create.sh]: removed commented out lines

File: kubernetes/samples/scripts/terraform/oke.create.sh (hunk 2 above)
Comment (marinakog) [reply kubernetes/samples/scripts/terraform/oke.create.sh]: fixed

File: kubernetes/samples/scripts/terraform/oke.create.sh (hunk 3 above)
Comment (marinakog) [reply kubernetes/samples/scripts/terraform/oke.create.sh]: fixed

File: kubernetes/samples/scripts/terraform/oke.delete.sh (hunk 4 above)
Comment (marinakog) [reply kubernetes/samples/scripts/terraform/oke.delete.sh]: fixed

File: oketest.sh (hunk 5 above)
Comment (marinakog) [reply oketest.sh:9]: fixed

File: oketest.sh (hunk 6 above)
Comment (marinakog) [reply oketest.sh]: fixed

File: oketest.sh (hunk 7 above)
Comment (marinakog) [reply oketest.sh]: fixed

File: oketest.sh (hunk 9 above)
Comment (marinakog) [reply oketest.sh:160]: fixed

File: oketest.sh (hunk 10 above)
Comment (marinakog) [reply oketest.sh]: fixed

File: oketest.sh (hunk 11 above)
Comment (marinakog) [reply oketest.sh]: fixed

File: oketest.sh (hunk 12 above)
Comment (marinakog) [reply oketest.sh]: done

File: kubernetes/samples/scripts/terraform/oke.create.sh [hunk 13]
```diff
@@ -68,6 +68,60 @@ function createCluster () {
...
+    NODES=`kubectl get nodes -o wide | grep "${privateIP}" | wc -l`
+    if [ "$NODES" == "1" ]; then
+      echo '- looks good'
+    else
+      echo '- could not talk to cluster, aborting'
+      cd ${terraformVarDir}
+      terraform destroy -auto-approve -var-file=${terraformVarDir}/${clusterTFVarsFile}.tfvars
+      exit 1
+    fi
+
```
Comment (xiancao) [inline kubernetes/samples/scripts/terraform/oke.create.sh:115]: delete this empty line

File: kubernetes/samples/scripts/terraform/oke.delete.sh [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+      for i in "${iparray[@]}"
+         do
+            lb=`oci lb load-balancer get --load-balancer-id=$i`
+            if [[ (-z "${lb##*$vcn_cidr_prefix*}") || (-z "${lb##*$k*}") ]] ;then
+               echo "deleting lb with id $i"
+               oci lb load-balancer delete --load-balancer-id=$i --force || true
+            fi
+        done
+    done
+
```
Comment (xiancao) [inline kubernetes/samples/scripts/terraform/oke.delete.sh:40]: delete this empty line

File: kubernetes/samples/scripts/terraform/oke.delete.sh [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+  terraform init -var-file=${terraform_script_dir}/${clusterName}.tfvars
+  terraform plan -var-file=${terraform_script_dir}/${clusterName}.tfvars
+  terraform destroy -auto-approve -var-file=${terraform_script_dir}/${clusterName}.tfvars
+}
+
+#MAIN
+oci_property_file=${1:-$PWD/oci.props}
+terraform_script_dir=${2:-$PWD}
+
+
```
Comment (xiancao) [inline kubernetes/samples/scripts/terraform/oke.delete.sh:54]: delete this empty line

File: kubernetes/samples/scripts/terraform/oke.delete.sh [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+clusterName=$(prop 'okeclustername')
+compartment_ocid=$(prop 'compartment.ocid')
+vcn_cidr_prefix=$(prop 'vcn.cidr.prefix')
+export KUBECONFIG=${terraform_script_dir}/${clusterName}_kubeconfig
+export PATH=${terraform_script_dir}/terraforminstall:$PATH
+
+
```
Comment (xiancao) [inline kubernetes/samples/scripts/terraform/oke.delete.sh:61]: delete this empty line

File: oketest.sh (hunk 9 above)
Comment (xiancao) [reply oketest.sh:160]: change this indentation to the same as the next line.

File: kubernetes/samples/scripts/terraform/oke.create.sh [hunk 17]
```diff
@@ -37,12 +37,12 @@ function setupTerraform () {
...
-       curl -O https://releases.hashicorp.com/terraform/0.11.8/terraform_0.11.8_linux_amd64.zip
-       unzip terraform_0.11.8_linux_amd64.zip
+       curl -LO --retry 3 https://releases.hashicorp.com/terraform/0.11.8/terraform_0.11.8_linux_amd64.zip
+       unzip -o terraform_0.11.8_linux_amd64.zip -d ${terraformDir}
     else
        echo "Unsupported OS"
     fi
-    chmod 777 ${terraformDir}/terraform
+    chmod +x ${terraformDir}/terraform
     export PATH=${terraformDir}:${PATH}
```
Comment (xiancao) [inline kubernetes/samples/scripts/terraform/oke.create.sh:47]: delete this empty line.

Comment (marinakog) [reply kubernetes/samples/scripts/terraform/oke.create.sh]: done

File: kubernetes/samples/scripts/terraform/oke.delete.sh (hunk 16 above)
Comment (marinakog) [reply kubernetes/samples/scripts/terraform/oke.delete.sh]: done

File: kubernetes/samples/scripts/terraform/oke.delete.sh (hunk 15 above)
Comment (marinakog) [reply kubernetes/samples/scripts/terraform/oke.delete.sh]: done

File: kubernetes/samples/scripts/terraform/oke.delete.sh (hunk 14 above)
Comment (marinakog) [reply kubernetes/samples/scripts/terraform/oke.delete.sh]: done

File: kubernetes/samples/scripts/terraform/oke.create.sh (hunk 13 above)
Comment (marinakog) [reply kubernetes/samples/scripts/terraform/oke.create.sh:111]: done

**Review by xiancao — APPROVED** [review]
LGTM

File: oketest.sh [hunk 18]
```diff
@@ -0,0 +1,183 @@
...
+if [[ -z "${WORKSPACE}" ]]; then
+  outdir="/scratch/${USER}/oketest"
+  export WORKSPACE=${PWD}
+else
+  outdir="${WORKSPACE}/logdir/${BUILD_TAG}"
+fi
+test_filter="**/It*"
+parallel_run="false"
+threads="2"
```
Comment (anpanigr) [inline oketest.sh:58]: Can we make the number of thread as variable when we run the tests in parallel.

Comment (marinakog) [reply oketest.sh:58]: it is optional variable, you can pass it as opt , this is just default value

File: oketest.sh [hunk 19]
```diff
@@ -0,0 +1,183 @@
...
+echo 'Clean up result root...'
+rm -rf "${RESULT_ROOT:?}/*"
+cd ${WORKSPACE}
+echo 'Run tests...'
+if [ "${maven_profile_name}" = "oke-cert" ]; then
+  echo "Running mvn -Dwdt.download.url=${wdt_download_url} -Dwit.download.url=${wit_download_url} -pl integration-tests -P ${maven_profile_name} verify"
+  mvn -Dwdt.download.url="${wdt_download_url}" -Dwit.download.url="${wit_download_url}" -Djdk.tls.client.protocols=TLSv1.2 -pl integration-tests -P ${maven_profile_name} verify 2>&1 | tee "${RESULT_ROOT}/oke.log"
+else
+  if [ "${parallel_run}" = "false" ]; then
+    echo "Running mvn -Dit.test=${test_filter} -Dwdt.download.url=${wdt_download_url} -Dwit.download.url=${wit_download_url} -pl integration-tests -P integration-tests verify"
```
Comment (vanajamukkara) [inline oketest.sh:177]: match with actual mvn command

File: oketest.sh [hunk 20]
```diff
@@ -0,0 +1,183 @@
...
+    mvn -Dit.test="${test_filter}, !ItExternalRmiTunneling, !ItSamples, !ItMiiSample, !ItTwoDomainsLoadBalancers, !ItMonitoringExporter, !ItPodRestart" -Dwdt.download.url="${wdt_download_url}" -Dwit.download.url="${wit_download_url}" -Djdk.tls.client.protocols=TLSv1.2 -pl integration-tests -P ${maven_profile_name} verify 2>&1 | tee "${RESULT_ROOT}/oke.log"
+  else
+    echo "Running mvn -Dit.test=${test_filter}, !ItOperatorUpgrade, !ItDedicatedMode -Dwdt.download.url=${wdt_download_url} -Dwit.download.url=${wit_download_url} -DPARALLEL_CLASSES=${parallel_run} -DNUMBER_OF_THREADS=${threads}  -pl integration-tests -P ${MVN_PROFILE} verify"
```
Comment (vanajamukkara) [inline oketest.sh:180]: match with actual mvn command

**Review by vanajamukkara — COMMENTED** [review]
LGTM

File: oketest.sh (hunk 19 above)
Comment (marinakog) [reply oketest.sh]: done

File: oketest.sh (hunk 20 above)
Comment (marinakog) [reply oketest.sh]: done

File: oketest.sh [hunk 21]
```diff
@@ -0,0 +1,183 @@
...
+
+export K8S_NODEPORT_HOST=$NODE_IP
+export JAVA_HOME="${JAVA_HOME:-`type -p java|xargs readlink -f|xargs dirname|xargs dirname`}"
+
+echo 'Clean up result root...'
+rm -rf "${RESULT_ROOT:?}/*"
+cd ${WORKSPACE}
+echo 'Run tests...'
+if [ "${maven_profile_name}" = "oke-cert" ]; then
+  echo "Running mvn -Dwdt.download.url=${wdt_download_url} -Dwit.download.url=${wit_download_url} -pl integration-tests -P ${maven_profile_name} verify"
```
[the remark this answers is not in the document]
Comment (marinakog) [reply oketest.sh]: done

**Review by vanajamukkara — APPROVED** [review]

**Pull request merged**